include/boost/lockfree/spsc_queue.hpp

```diff
@@ -761,6 +856,41 @@ class spsc_queue:
         return base_type::push(t);
     }
 
+#ifndef BOOST_NO_CXX11_RVALUE_REFERENCES
+
+    /** Pushes object t to the ringbuffer via move construction/
+     *
+     * \pre only one thread is allowed to push data to the spsc_queue
+     * \post object will be pushed to the spsc_queue, unless it is full.
+     * \return true, if the push operation is successful.
+     *
+     * \note Thread-safe and wait-free
```

Comment:
maybe comment about the API which is not available if `BOOST_NO_CXX11_RVALUE_REFERENCES` is defined